### PR TITLE
Implement `GetEpochSyncCommittees` in the beacon API

### PR DESCRIPTION
**Pull request by rkapka**
Merge to `develop` equivalent of #9396.

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]
Refactored for single responsibilities and more coverage

File: beacon-chain/rpc/eth/beacon/sync_committee.go [hunk 1]
```diff
@@ -20,64 +18,82 @@ import (
...
+func (bs *Server) SubmitSyncCommitteeSignature(ctx context.Context, message *eth.SyncCommitteeMessage) (*empty.Empty, error) {
+	panic("implement me")
+}
+
+func currentCommitteeIndicesFromState(st state.BeaconState) ([]types.ValidatorIndex, *ethpb.SyncCommittee, error) {
 	committee, err := st.CurrentSyncCommittee()
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not get sync committee: %v", err)
+		return nil, nil, status.Errorf(
+			codes.Internal, "Could not get sync committee: %v", err,
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/sync_committee.go:55]: I think we should return regular errors from helper functions and `status` errors only from top-level functions.

File: beacon-chain/rpc/eth/beacon/sync_committee.go [hunk 2]
```diff
@@ -20,64 +18,82 @@ import (
...
 	committeeIndices := make([]types.ValidatorIndex, len(committee.Pubkeys))
 	for i, key := range committee.Pubkeys {
 		index, ok := st.ValidatorIndexByPubkey(bytesutil.ToBytes48(key))
 		if !ok {
-			return nil, status.Errorf(codes.Internal, "Validator index not found for pubkey %#x", bytesutil.Trunc(key))
+			return nil, nil, status.Errorf(
+				codes.Internal,
+				"Validator index not found for pubkey %#x",
+				bytesutil.Trunc(key),
+			)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/sync_committee.go:67]: Same here.

File: beacon-chain/rpc/eth/beacon/sync_committee.go [hunk 3]
```diff
@@ -20,64 +18,82 @@ import (
...
+func extractSyncSubcommittees(st state.BeaconState, committee *ethpb.SyncCommittee) ([]*eth.SyncSubcommitteeValidators, error) {
 	subcommitteeCount := params.BeaconConfig().SyncCommitteeSubnetCount
 	subcommittees := make([]*eth.SyncSubcommitteeValidators, subcommitteeCount)
 	for i := uint64(0); i < subcommitteeCount; i++ {
 		pubkeys, err := altair.SyncSubCommitteePubkeys(committee, types.CommitteeIndex(i))
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, "Failed to get subcommittee pubkeys: %v", err)
+			return nil, status.Errorf(
+				codes.Internal, "Failed to get subcommittee pubkeys: %v", err,
+			)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/sync_committee.go:82]: Same here.

File: beacon-chain/rpc/eth/beacon/sync_committee.go [hunk 4]
```diff
@@ -20,64 +18,82 @@ import (
...
 		subcommittee := &eth.SyncSubcommitteeValidators{Validators: make([]types.ValidatorIndex, len(pubkeys))}
 		for j, key := range pubkeys {
 			index, ok := st.ValidatorIndexByPubkey(bytesutil.ToBytes48(key))
 			if !ok {
-				return nil, status.Errorf(codes.Internal, "Validator index not found for pubkey %#x", bytesutil.Trunc(key))
+				return nil, status.Errorf(
+					codes.Internal,
+					"Validator index not found for pubkey %#x",
+					bytesutil.Trunc(key),
+				)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/sync_committee.go:92]: Same here.

**Review by rauljordan — APPROVED** [review]
Suggested accepted by Radek off Github

**Pull request merged**